src/Block/Service/EditableBlockService.php

```diff
@@ -29,10 +29,5 @@ public function configureCreateForm(FormMapper $form, BlockInterface $block);
 
     public function validate(ErrorElement $errorElement, BlockInterface $block);
 
-    /**
-     * @param string|null $code
-     *
-     * @return MetadataInterface
-     */
-    public function getMetadata($code = null);
+    public function getMetadata(?string $code = null): MetadataInterface;
```

Comment:
Is the `?` modifier required if this argument already supports `null` as value via its default value?

Reply:
It is, because implementors might use `string $code = ''`, and that would make it no longer `nullable`. Now they are forced to support `null`

Reply:
I'm not sure if we even need this parameter...

We use this parameter as a fallback for the descriptions which looks wrong
https://github.com/sonata-project/SonataNewsBundle/blob/3.x/src/Block/RecentPostsBlockService.php#L152-L157

Reply:
We should deprecate this, even if we have no running implementation of this interface

Reply:
Should I just remove it? It's still compatible with implementations that have a this be a default argument: https://3v4l.org/lIYFt

Reply:
Yes, please remove this 👍